components/eam/src/physics/cam/zm_conv.F90

```diff
@@ -916,10 +1718,87 @@ subroutine zm_convr(lchnk   ,ncol    , &
 ! Treat rliq as flux out bottom, to be added back later.
    do k = 1, pver
       do i = 1, ncol
-         rliq(i) = rliq(i) + dlf(i,k)*dpp(i,k)/gravit
+          rliq(i) = rliq(i) + (dlf(i,k)+dif(i,k)+dsf(i,k))*dpp(i,k)/gravit
+          rice(i) = rice(i) + (dif(i,k)+dsf(i,k))*dpp(i,k)/gravit
       end do
    end do
    rliq(:ncol) = rliq(:ncol) /1000._r8
+   rice(:ncol) = rice(:ncol) /1000._r8    
+
+   if (zm_microp) then
+     deallocate( &
```

Comment:
see comment above about moving this to it's own routine in the appropriate module

Reply:
Deallocations are moved into corresponding subroutine.